libr/cons/cons.c

```diff
@@ -687,21 +687,44 @@ R_API void r_cons_context_break(RConsContext *context) {
 	}
 }
 
-R_API void r_cons_flush() {
+#define CTX(x) I.context->x
+
+R_API void r_cons_last(void) {
+	CTX(lastMode) = true;
+	r_cons_memcat (CTX(lastOutput), CTX(lastLength));
+	//r_cons_print (lastOutput);
+}
+
+R_API void r_cons_flush(void) {
 	const char *tee = I.teefile;
-	if (I.noflush) {
+	if (I.noflush || I.context->buffer_len < 1) {
 		return;
 	}
 	if (I.null) {
 		r_cons_reset ();
 		return;
 	}
+	if (!I.filter && I.context->grep.nstrings < 1 && \
+		!I.context->grep.tokens_used && !I.context->grep.less && \
+		!I.context->grep.json && !I.is_html) {
+		if (!CTX(lastMode)) {
+			// snapshot of the output
+			free (CTX(lastOutput));
+			CTX(lastOutput) = calloc (CTX(buffer_len) + 1, 1);
+			if (CTX(lastOutput)) {
+				CTX(lastLength) = CTX(buffer_len);
+				memcpy (CTX(lastOutput), CTX(buffer), CTX(buffer_len));
+			}
+		}
+		CTX(lastMode) = false;
```

Comment:
For the else-branch the assignment seems trivial. So it actually belongs to the if-branch?

Reply:
Agree. Will change before merge